tee-worker/service/src/main.rs

```diff
@@ -872,3 +828,47 @@ fn check_we_are_primary_validateer(
 		node_api.enclave_count(Some(*register_enclave_xt_header.parent_hash()))?;
 	Ok(enclave_count_of_previous_block == 0)
 }
+
+fn data_provider(config: &Config) -> DataProvidersStatic {
+	let build_in_mode = vec!["dev", "mock", "prod", "staging"];
+	let build_in_config: Value =
+		serde_json::from_slice(include_bytes!("running-mode-config.json")).unwrap();
+
+	let mut data_provider_config = if build_in_mode.contains(&config.running_mode.as_str()) {
+		let config = build_in_config.get(config.running_mode.as_str()).unwrap();
+		serde_json::from_value::<DataProvidersStatic>(config.clone()).unwrap()
+	} else {
+		let file_path = config.running_mode.as_str();
+		let mut file = File::open(file_path)
+			.map_err(|e| format!("{:?}, file:{}", e, file_path))
+			.unwrap();
+		let mut data = String::new();
+		file.read_to_string(&mut data).unwrap();
+		serde_json::from_str::<DataProvidersStatic>(data.as_str()).unwrap()
+	};
+	if let Ok(v) = env::var("TWITTER_OFFICIAL_URL") {
```

Comment:
A small question is if all `env`s are set but the running-mode is incorrectly configured, should we still allow this? Current logic is we would panic out before checking the env.

Not a big deal though

Reply:
I think `built_in_config` or config from file_path should always be correct. :sweat_smile: